plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/validation/RequestMarkersHandler.java

```diff
@@ -43,27 +42,24 @@ public class RequestMarkersHandler extends AbstractActionHandler<RequestMarkersA
    @Override
    @SuppressWarnings("checkstyle:cyclomaticComplexity")
    public List<Action> executeAction(final RequestMarkersAction action) {
-      List<String> elementsIDs = action.getElementsIDs();
+      List<String> elementIds = action.getElementsIDs();
       if (validator.isEmpty()) {
          LOGGER.warn("Cannot compute markers! No implementation for " + ModelValidator.class + " has been bound");
          return none();
       }
 
       // if no element ID is provided, compute the markers for the complete model
-      if (elementsIDs == null || elementsIDs.size() == 0
-         || (elementsIDs.size() == 1 && "EMPTY".equals(elementsIDs.get(0)))) {
-         elementsIDs = Arrays.asList(modelState.getRoot().getId());
+      if (elementIds == null || elementIds.size() == 0
+         || (elementIds.size() == 1 && "EMPTY".equals(elementIds.get(0)))) {
+         elementIds = Arrays.asList(modelState.getRoot().getId());
       }
-      List<Marker> markers = new ArrayList<>();
-      GModelIndex currentModelIndex = modelState.getIndex();
-      for (String elementID : elementsIDs) {
-         Optional<GModelElement> modelElement = currentModelIndex.get(elementID);
-         if (modelElement.isPresent()) {
-            markers.addAll(validator.get().validate(modelElement.get()));
-         }
 
-      }
+      String reason = action.getReason() != null ? action.getReason() : MarkersReason.BATCH;
```

Comment:
See comment above. If the default is already properly set in the action class we don't need to handle it explicitly here.